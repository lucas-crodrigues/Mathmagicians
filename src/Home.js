import Header from './components/Header';
import Footer from './components/Footer';

const Home = () => (
  <div
    className="home"
    style={{
      width: '100vw',
      height: '100vh',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'space-between',
    }}
  >
    <Header />
    <div>
      <h3
        style={{
          margin: '4vh 2vw',
        }}
      >
        Welcome to our page!
      </h3>
      <p
        style={{
          margin: '4vh 2vw',
        }}
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        Aenean blandit, erat eget hendrerit gravida, ligula velit aliquet nisi,
        in placerat dui elit bibendum justo.
        Nulla a ipsum blandit, congue lacus in, sagittis mauris.
        Nullam placerat auctor ex. Ut a enim vitae lacus euismod tristique.
        Phasellus nec semper magna, facilisis malesuada purus.
        Morbi tincidunt vel nibh ut vehicula. Etiam vel neque orci.
        <br />
        Aenean quis enim vel nisl feugiat hendrerit. Maecenas ac faucibus magna.
        Quisque condimentum nibh a eros posuere, fringilla rutrum magna sagittis.
        Praesent eleifend volutpat libero ut mollis. Nam sit amet arcu est.
        Mauris at auctor sapien. Nunc volutpat aliquet facilisis. Morbi a interdum magna.
      </p>
    </div>
    <Footer />
  </div>
);

export default Home;
