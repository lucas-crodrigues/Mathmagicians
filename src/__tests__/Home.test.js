import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../Home';

describe('Home page test suites', () => {
  test('Home page snapshot', () => {
    const renderedComponent = renderer.create(<BrowserRouter><Home /></BrowserRouter>).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test('Check home page title', () => {
    render(<BrowserRouter><Home /></BrowserRouter>);
    const homeHeading = screen.getByRole('heading', { name: /welcome to our page!/i });
    expect(homeHeading).toHaveTextContent('Welcome to our page!');
  });
});
