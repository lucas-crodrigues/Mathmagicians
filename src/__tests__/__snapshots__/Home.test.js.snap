// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test suites Home page snapshot 1`] = `
<div
  className="home"
>
  <header
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderBottom": "1px solid gray",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "4vh 2vw",
      }
    }
  >
    <h1
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      Math Magicians
    </h1>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-around",
          "width": "25vw",
        }
      }
    >
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </nav>
      <div
        style={
          Object {
            "borderLeft": "1px solid gray",
            "height": "5vh",
          }
        }
      />
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </nav>
      <div
        style={
          Object {
            "borderLeft": "1px solid gray",
            "height": "5vh",
          }
        }
      />
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </nav>
    </div>
  </header>
  <div
    style={
      Object {
        "paddingBottom": "20vh",
      }
    }
  >
    <h3
      style={
        Object {
          "fontSize": "30px",
          "margin": "4vh 2vw",
        }
      }
    >
      Welcome to our page!
    </h3>
    <p
      style={
        Object {
          "margin": "4vh 3vw",
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent lobortis urna quis metus interdum, sit amet tincidunt quam fringilla. Sed interdum est ornare ipsum faucibus, vitae aliquam nibh sodales. Aenean vestibulum nisl sit amet nunc aliquet blandit. Donec posuere enim et augue varius, eu pulvinar nunc aliquet. Nulla id blandit quam. Etiam porta laoreet nisl et laoreet. Pellentesque mollis quis ex consequat interdum. Morbi sodales lorem non neque consequat molestie. Donec finibus felis ut justo scelerisque gravida. Integer vel lectus sit amet ipsum pharetra pellentesque.
      <br />
      Nulla nibh lectus, blandit ut sem vitae, consectetur imperdiet erat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec euismod scelerisque odio, at interdum arcu laoreet vitae. Duis maximus mauris risus, rhoncus iaculis nibh pretium sit amet. Nunc quam libero, porttitor sed turpis id, cursus dictum elit. Aenean eu ultricies ligula, in porta dolor. Aenean posuere eleifend urna commodo tempor. Nulla condimentum leo est, in elementum augue vestibulum nec. Vestibulum fringilla vel sem in lobortis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. In mattis lobortis augue, vitae blandit est varius eu.
      <br />
      <br />
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent lobortis urna quis metus interdum, sit amet tincidunt quam fringilla. Sed interdum est ornare ipsum faucibus, vitae aliquam nibh sodales. Aenean vestibulum nisl sit amet nunc aliquet blandit. Donec posuere enim et augue varius, eu pulvinar nunc aliquet. Nulla id blandit quam. Etiam porta laoreet nisl et laoreet. Pellentesque mollis quis ex consequat interdum. Morbi sodales lorem non neque consequat molestie. Donec finibus felis ut justo scelerisque gravida. Integer vel lectus sit amet ipsum pharetra pellentesque.
      <br />
      Nulla nibh lectus, blandit ut sem vitae, consectetur imperdiet erat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec euismod scelerisque odio, at interdum arcu laoreet vitae. Duis maximus mauris risus, rhoncus iaculis nibh pretium sit amet. Nunc quam libero, porttitor sed turpis id, cursus dictum elit. Aenean eu ultricies ligula, in porta dolor. Aenean posuere eleifend urna commodo tempor. Nulla condimentum leo est, in elementum augue vestibulum nec. Vestibulum fringilla vel sem in lobortis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. In mattis lobortis augue, vitae blandit est varius eu.
      <br />
    </p>
  </div>
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "borderTop": "1px solid black",
        "height": "25px",
      }
    }
  />
</div>
`;
